### PR TITLE
Nerfs KAs in Atmosphere

### PR DESCRIPTION
**Pull request by CampinKiller24**
KAs have long been considered a strong weapon in low/no atmosphere only, despite this not being the case, especially with the right upgrades. A "properly" upgraded KA can instantly crit someone in heavy armor in a single shot in atmosphere currently. This nerfs KAs in atmosphere (>~40 kPA), but keeps them strong in low atmosphere/void conditions, as was intended. Also nerfs a couple components that allow straight shredding of people in armor.

EDIT: For reference, a "fully upgraded" KA would now do the following damages:
Unarmored target in atmosphere: ~25 brute damage
Target in heavy armor in atmosphere: ~16 brute damage
Target in security voidsuit in void: ~65 brute damage

### CODE REVIEW

**Comment by CampinKiller24** [comment]
!review

**Review by NonQueueingMatt — APPROVED** [review]

File: code/modules/custom_ka/core.dm [hunk 1]
```diff
@@ -234,19 +234,43 @@
...
+		
+	//Get turf and see if in atmos or not
+	var/turf/T = get_turf(src)
+	
+	if(T)
+		var/datum/gas_mixture/environment = T.return_air()
+		var/pressure = (environment)? environment.return_pressure() : 0
+		//If pressure is greater than about 40 kPA
+		if(pressure > ONE_ATMOSPHERE*0.4)
+			if(ispath(installed_barrel.projectile_type, /obj/item/projectile/kinetic))
```
Comment (NonQueueingMatt) [inline code/modules/custom_ka/core.dm:246]: istype

File: code/modules/custom_ka/core.dm [hunk 2]
```diff
@@ -234,19 +234,43 @@
...
+		if(pressure < ONE_ATMOSPHERE*0.4)
+			if(ispath(installed_barrel.projectile_type, /obj/item/projectile/kinetic))
+				var/obj/item/projectile/kinetic/shot_projectile = new installed_barrel.projectile_type(get_turf(src))
+				shot_projectile.damage = damage_increase
+				shot_projectile.range = range_increase
+				shot_projectile.aoe = max(1, aoe_increase)
+				shot_projectile.base_damage = damage_increase
+				return shot_projectile
+				
+			if(ispath(installed_barrel.projectile_type, /obj/item/projectile/beam))
```
Comment (NonQueueingMatt) [inline code/modules/custom_ka/core.dm:269]: istype

File: code/modules/custom_ka/core.dm [hunk 3]
```diff
@@ -234,19 +234,43 @@
...
+	
+	if(T)
+		var/datum/gas_mixture/environment = T.return_air()
+		var/pressure = (environment)? environment.return_pressure() : 0
+		//If pressure is greater than about 40 kPA
+		if(pressure > ONE_ATMOSPHERE*0.4)
+			if(ispath(installed_barrel.projectile_type, /obj/item/projectile/kinetic))
+				var/obj/item/projectile/kinetic/shot_projectile = new installed_barrel.projectile_type(get_turf(src))
+				shot_projectile.damage = damage_increase
+				shot_projectile.range = range_increase
```
Comment (NonQueueingMatt) [inline code/modules/custom_ka/core.dm:249]: this is a lot of copypaste, do this on a generic projectile typecast, then edit the  other types as needed

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

**Pull request closed without merging**